Update footer to match new mockups

obulat opened the pull request:
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->
Fixes #2012 by @obulat
Related to #2009

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->
This PR updates the footer to match the latest mockups.

The footer layout should be updated to match the screenshot.

## Mockup 
- Footer component: [Design Library, upcoming footer page](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=3018%3A7629&t=gjE9wqTv3hxc2Kek-0).

## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->
The footer should work properly, and match the mockups.

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

zackkrida commented:
@obulat should there be accompanying updates to visual regression tests here? Thanks for the speedy implementation!

dhruvkb commented on src/components/VFooter/VFooter.vue:
```diff
@@ -113,52 +115,91 @@ export default defineComponent({
+  width: 100% !important;
+}
+.footer-sm .logo-and-links {
+  display: grid;
+  grid-auto-flow: column;
+  grid-template-columns: repeat(2, 1fr);
```
IMO these CSS rules should be replaced `@apply`-ied Tailwind utilities.

obulat replied:
Done :)

obulat commented on src/components/VFooter/VFooter.vue:
```diff
@@ -113,52 +115,91 @@ export default defineComponent({
+  width: 100% !important;
+}
+.footer-sm .logo-and-links {
+  display: grid;
+  grid-auto-flow: column;
+  grid-template-columns: repeat(2, 1fr);
```
```suggestion
  @apply grid grid-flow-col grid-cols-1;
```

obulat commented on src/components/VFooter/VFooter.vue:
```diff
@@ -99,66 +101,99 @@ export default defineComponent({
-        )
-      }
-    })
+    const isContentMode = computed(() => props.mode === 'content')
```
The wrapper component should know whether the footer is in 'content' mode or in the 'internal' mode.

panchovm reviewed (changes requested):
It looks great, I just noticed the following details.

## Content version
### Spacings
* In `xl` and `2xl` the left and right padding should be `40px` instead of `24px`
* In `lg`, `xl`, and `2xl`, the gap between rows should be `32px` instead of `40px`

### Components
For the page links, I think you used the "Clear" component from the drawer. In theory, that is correct, but the element has a fixed height to match the spacing rule of 8x.

<img width="417" alt="CleanShot 2022-12-01 at 14 26 24@2x" src="https://user-images.githubusercontent.com/895819/205064541-cdbde9bd-0f16-4086-a3d5-2affe2e63edf.png">

The element has a decimal size of `32.9`, which can be problematic and break layouts if we keep using it as it is. We can tackle this in two ways I think:
1. Replace the Clear component with the [Link component](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=3667%3A9753&t=LBXemTz9meYwiYiy-4) and apply the spacing rule to the gap between `li` elements
2. Set a fixed height of the Clear component as `32px` and add a new spacing value between `li` to match the mockup.

I am drawn to the first option as the **Link** component was thought for links single links, whereas the **Clear** component inherits the fixed height from the Results button. Why did this happen? Because the link component did not exist at the moment I started the header redesign.

In `sm`, the locale component has a `max-width: 100%` that crashes the layout. The mockup keeps the width fixed as `230px` in `sm, md, lg, xl, 2xl` to avoid this effect. Is that possible or better to find a different solution?

<img width="564" alt="CleanShot 2022-12-01 at 14 38 33@2x" src="https://user-images.githubusercontent.com/895819/205067258-87e3b103-381d-4fc9-82ee-43696c8b7cb1.png">

### Alignment
In `sm` and `md`, the locale component and WordPress call need to be aligned vertically centered.

<img width="745" alt="CleanShot 2022-12-01 at 14 35 32@2x" src="https://user-images.githubusercontent.com/895819/205066572-67635f67-3985-4837-bbdd-854baacd40ad.png">

## Internal version
### Spacings
* In `xl` and `2xl` the left and right padding should be `40px` instead of `24px`
* In `sm` and `md`, padding values should be `24px, 24px, 40px, 24px`. The top part is bigger to match the result area layout.

### Alignment
* In `sm` and `md`, the WordPress call should be fixed on the right side.
* In `sm` and `md`, the locale component and WordPress call need to be aligned vertically centered.

panchovm reviewed (changes requested):
It looks great. Just a few notes

## Content version
* Top and bottom paddings must be `40px` in all breakpoints.

## Internal version
* In `lg, xl, 2xl` all sides should be `40px`. I see a `24px` for the bottom side.
* In `sm, md` the top and bottom spacings are inverted. It should be `40px` for the bottom and `24px` for the top.

## Nav items
Despite the above changes. Which one do you think is the best option to address the nav items' components?

obulat commented:
Thank you for the review, @panchovm! I've fixed the footer paddings.

> Despite the above changes. Which one do you think is the best option to address the nav items' components?

> The element has a decimal size of 32.9, which can be problematic and break layouts if we keep using it as it is.

What kind of problems will the decimal size of 32.9 bring? I don't think we can set a fixed height in the navigation bar because we need the designs to be used in other languages, too. Some navigation items can have a long title when translated, and then they will not fit into the component and will overflow. 

So, we have to set the height using text height + padding. The text height, when it's in one line, is decimal, 16.5px. We either need to set a decimal padding to get a nice link height of 32px, or we have to live with a decimal height of the link:
<img width="282" alt="Screenshot 2022-12-07 at 9 11 11 AM" src="https://user-images.githubusercontent.com/15233243/206104573-4fe0ba10-24ba-4561-a2d4-4fedf24b28e7.png">

In the code, we don't really have a `Clear` component, do we have it in the Figma mockups? 
I have added a `VNavLink` component for navigation links. It has an 8px vertical padding to ensure that the links are tappable. I also added some row and column gaps for the footer.

This is what the height becomes for one-line items:
<img width="456" alt="Screenshot 2022-12-07 at 9 17 04 AM" src="https://user-images.githubusercontent.com/15233243/206104555-2b0c888b-a6b3-4aeb-8981-51a31a0e24ce.png">

And here, what items with more than one-line of text will look like:
<img width="528" alt="Screenshot 2022-12-07 at 9 40 10 AM" src="https://user-images.githubusercontent.com/15233243/206107115-72a61424-1e5e-48a2-a8e2-214669fb8237.png">

Is this approach OK, or should we think of something else, @panchovm?

panchovm commented:
> I don't think we can set a fixed height in the navigation bar because we need the designs to be used in other languages

Very much agree with this. My point was about the need for `8px` of padding as the focus style looks strange.

<img width="353" alt="CleanShot 2022-12-07 at 19 16 29@2x" src="https://user-images.githubusercontent.com/895819/206263469-26ef19ce-13bf-41ec-a1e7-8b1ecee426d8.png">

Your point about the tappable area makes sense, but I think we don't need it as the [Target Size Criterion](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html) requires _at least_ a `44px` area.

What you did is definitely correct. I am just bringing this up to reach consistency across other similar elements. I am thinking of the "Clear" action in filters, "Report this content" in the single result, these nav links in the Footer, and possibly others that I am missing.

The `Clear` component does exist in Figma, but since it does not in code, and we are discussing its purpose, it seems timely to define a basic link element that does not have paddings/margins, could have an icon, and where focus ring wraps the element. Something like the image below, but called `VSingleLink` instead (name TBD).

<img width="300" alt="CleanShot 2022-12-07 at 19 24 54@2x" src="https://user-images.githubusercontent.com/895819/206265159-76b8e9eb-a6e3-4993-8948-dd59235c5708.png">

What do you think?

obulat commented:
> Your point about the tappable area makes sense, but I think we don't need it as the [Target Size Criterion](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html) requires _at least_ a `44px` area.

Today I learned: it's ok for the tappable area (or the target size on the desktop) to be smaller than 44px if the areas do not overlap. So, it's OK from the accessibility point of view (or SEO in lighthouse checks) if the focus area of the links is small, like in the snapshots you shared, @panchovm, but there is enough space between the links. The problem starts when the links are so close to each other that a click on one can be interpreted as a click on another one. And we don't have such problem anyways :) 

I've updated the code to have a small focus "box" around the links. The padding now is on the `<li>` element.

panchovm reviewed (approved):
It looks great. Nice work ✨

zackkrida commented on src/components/VHeader/VWordPressLink.vue:
```diff
@@ -0,0 +1,36 @@
+    >
+      <template #wordpress>
+        <WordPress class="aria-hidden" />
+        <span class="sr-only">WordPress</span>
```
Great inclusion for a11y.

zackkrida reviewed (approved):
Looks good @obulat!
